arch/x86/soc/intel_quark/quark_d2000/Kconfig.defconfig.series

```diff
@@ -99,6 +99,7 @@ config I2C_QMSI
 if I2C_QMSI
 config I2C_0
 	def_bool y
+
 config I2C_0_IRQ_PRI
```

Comment:
I am sure that I am missing something here. There is no interrupt property in intel,qmsi-i2c but there is IRQ_PRI here.

Reply:
Indeed, sounds like some old left-over.

Reply:
Actuall, we both overlooked: it is there. Difference being that irq connection is done in init function.

Reply:
I was talking about intel,qmsi-i2c.dtsi file. Will you add support to the i2c_qmsi driver to get IRQ configuration from DT.

Reply:
aah, ok. I tried that but interrupts can't take more than 2 arguments there. Adding a interrupt-priority attribute could work, but not nice.

Reply:
Did you try to change #interrupt-cells number in intc node?

Reply:
Ah, that could help, thanks. I am new to dts, thus the trial & error path...

Reply:
Does not work, because of the interrupt-parent, &intc sets the interrupt-cells to 2.

Reply:
I meant to change &intc node the interrupt-cells to 3 and update all the nodes using it (uart, i2c spi, ...).

Reply:
oh, true. I need a break.

Will see that.

Reply:
Does not work, seems like a bug in extract_dts_includes.py to me but I am not entirely sure. (this tool _really_ needs to be rewritten)

Reply:
I think you should modify also intel,ioapic.yaml from:
```
cell_string: IRQ

"#cells":
  - irq
  - sense
```
to:
```
cell_string: IRQ

"#cells":
  - irq
  - priority
  - sense
```
but I am not sure.

Reply:
Does the intel,ioapic support priorities?  From what I can tell it doesn't seem to.  The binding for ioapic yaml is based on the linux binding if I remember correctly.

Reply:
It does support priorities. I don't know why Linux does not expose this setting, could it be because the intel platforms supported there provide all the necessary information through PCI?

Reply:
Looking at the driver in zephyr, it doesn't seem that we allow or have means to set the priority in the ioapic (but I could be missing something).

Reply:
I am not really aware of the details in that part, but it seems that ioapic itself does not manage priorities directly, hardware wise. It seems to be handled in software through the ISR table.

Reply:
We *do* set priorities. But it is not done in the IOAPIC. Priorities are reflected by the vector assignment in the IDT. Based on the priority level given to IRQ_CONNECT(), gen_idt uses this to find an appropriate range in the vector table to configure the interrupt. So the DTS configuration will need to express priorities.

Reply:
> We do set priorities. But it is not done in the IOAPIC. Priorities are reflected by the vector assignment in the IDT. Based on the priority level given to IRQ_CONNECT(), gen_idt uses this to find an appropriate range in the vector table to configure the interrupt. So the DTS configuration will need to express priorities.

Trying to understand this, based on include/arch/x86/arch.h - it looks like MVIC doesn't support priority, and ioapic is done by how vector values are assigned.  Is that correct?